be/btree.c

```diff
@@ -488,12 +488,14 @@ static void be_btree_split_child(struct m0_be_btree *btree,
 	/* Copy the latter half keys from the current child to the new child */
 	i = 0;
 	while (i < new_child->bt_num_active_key) {
-		new_child->bt_kv_arr[i] = child->bt_kv_arr[i + BTREE_FAN_OUT];
-		*new_child->bt_ik[i].inlkey = *((char *)child->bt_kv_arr[i + BTREE_FAN_OUT].btree_key);
-		new_child->allocated[i] = 1;
+		new_child->bt_kv_arr[i] = child->bt_kv_arr[i + BTREE_FAN_OUT];		
 		if(btree->bb_ops->ko_type != M0_BBT_CAS_CTG)
 		{
-			new_child->bt_kv_arr[i].btree_key = &new_child->bt_ik[i].inlkey;
+			int tindex = find_next_avaliable_index(new_child->allocated);
+			memcpy(new_child->bt_ik[tindex].inlkey,child->bt_kv_arr[i + BTREE_FAN_OUT].btree_key,64);
+			new_child->allocated[tindex] = 1;
+			new_child->bt_kv_arr[i].btree_key = &new_child->bt_ik[tindex].inlkey;
+			child->allocated[i + BTREE_FAN_OUT] = 0;
```

Comment:
This line is not correct. You need to exactly find the allocated offset which needs to be set to '0'.